authentication/php/phpleague/src/Oauth/OnBehalfOfCertificateContext.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+
+namespace Microsoft\Kiota\Authentication\Oauth;
+
+/**
+ * Class OnBehalfOfCertificateContext
+ *
+ * on_behalf_of flow using certificate
+ *
+ * @package Microsoft\Kiota\Authentication\Oauth
+ * @copyright 2022 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+class OnBehalfOfCertificateContext extends BaseCertificateContext
+{
+    /**
+     * @var string
+     */
+    private string $assertion;
+    /**
+     * @var array
+     */
+    private array $additionalParams;
+
+    /**
+     * @param string $tenantId
+     * @param string $clientId
+     * @param string $assertion
+     * @param string $certificatePath
+     * @param string $certificatePassphrase
+     * @param array $additionalParams<string, string>
+     */
+    public function __construct(string $tenantId, string $clientId, string $assertion, string $certificatePath, string $certificatePassphrase = '', array $additionalParams = [])
```

Comment:
Yeah this is going to be a limitation for server applications, which is all applications in the case of PHP. But If we look at Azure Identity they have the same limitation (java in my previous comment, dotnet here https://docs.microsoft.com/en-us/dotnet/api/azure.identity.onbehalfofcredential.-ctor?view=azure-dotnet#azure-identity-onbehalfofcredential-ctor(system-string-system-string-system-string-system-string))
I think we should leave it as a string for now, knowing it might push people to new up their request adapter & client more often than they would with a callback pattern, see if we get any demand, and if we do, try to push Azure Identity to make a PHP lib. Thoughts?

Reply:
Sounds good to me